src/execution_plan.c

```diff
@@ -1812,7 +1822,7 @@ int ExecutionPlan_ExecutionGet(RedisModuleCtx *ctx, RedisModuleString **argv, in
             ".collect()"
             ".flatmap(lambda x: [i for i in x])"
             ".run(convertToStr=False, collect=False)", eid);
-        int res = RedisGearsPy_ExecuteReturnResultsOnly(ctx, fargv, 2);
+        int res = RedisGearsPy_ExecuteWithCallback(ctx, fargv, 2, onDoneResultsOnly);
         RedisModule_Free(fargv[1]);
```

Comment:
The only thing I do not like here is that you have to create it fargv with size of 2 (cause the function assume it also contains the command name). Lets add a todo here and leave it like this for now.

Reply:
Added TODO in f8911ec